### PR TITLE
feat(condo): DOMA-6404 scope type was added

### PR DESCRIPTION
**Pull request by AleX83Xpert**
Also, no scopes - no access! To create access for all organization you must create empty scope (no property, no unitType, no unitName)

### CODE REVIEW

File: apps/condo/domains/news/access/NewsItemScope.js [hunk 1]
```diff
@@ -41,17 +41,14 @@ async function canManageNewsItemScopes ({ authentication: { item: user }, origin
...
     if (user.type === STAFF) {
-        let newsItemId = get(originalInput, ['newsItem', 'connect', 'id'])
         if (operation === 'update') {
-            if (!itemId) return false
-
-            if (!newsItemId) {
-                const newsItemScope = await getById('NewsItemScope', itemId)
-                newsItemId = newsItemScope.newsItem
-            }
+            return false
```
Comment (pahaz) [inline apps/condo/domains/news/access/NewsItemScope.js:45]: you can change the unitName if you want? (optional)

File: apps/condo/domains/news/schema/NewsItemScope.js [hunk 2]
```diff
@@ -78,17 +80,30 @@ const NewsItemScope = new GQLListSchema('NewsItemScope', {
 
     },
     hooks: {
-        validateInput: async (args) => {
-            const { resolvedData, existingItem, context, operation } = args
-            const possibleItemData = { ...existingItem, ...resolvedData }
-
-            if (!get(possibleItemData, 'property') && !get(possibleItemData, 'unitType') && !get(possibleItemData, 'unitName')) {
-                throw new GQLError(ERRORS.EMPTY_NEWS_ITEM_SCOPE, context)
+        resolveInput: async ({ operation, existingItem, resolvedData }) => {
```
Comment (pahaz) [inline apps/condo/domains/news/schema/NewsItemScope.js:83]: it's better to move this resolve inside field

Comment (pahaz) [reply apps/condo/domains/news/schema/NewsItemScope.js:83]: or at least group this logic inside some function

**Review by pahaz — COMMENTED** [review]
What about constraints for scopes

File: apps/condo/domains/news/access/NewsItemScope.js (hunk 1 above)
Comment (AleX83Xpert) [reply apps/condo/domains/news/access/NewsItemScope.js:45]: If receivers are changed, we create a new set of scopes.

File: apps/condo/domains/news/schema/NewsItemScope.test.js [hunk 3]
```diff
@@ -292,29 +314,11 @@ describe('NewsItemScope', () => {
...
-            await expectToThrowGQLError(
+            await expectToThrowValidationFailureError(
                 async () => await createTestNewsItemScope(adminClient, newsItem, { unitName: '1' }),
-                {
-                    code: 'BAD_USER_INPUT',
-                    type: 'UNIT_NAME_WITHOUT_UNIT_TYPE',
-                    message: 'You set unitName without unitType',
-                    messageForUser: 'api.newsItem.UNIT_NAME_WITHOUT_UNIT_TYPE',
-                },
+                'Required field "type" is null or undefined.',
```
Comment (pahaz) [inline apps/condo/domains/news/schema/NewsItemScope.test.js:321]: it looks like wrong error name

File: apps/condo/domains/news/utils/accessSchema/index.js [hunk 4]
```diff
@@ -14,6 +14,13 @@ function queryFindNewsItemsScopesByResidents (residents) {
             return [
                 {
                     OR: [
+                        {
+                            AND: [
```
Comment (pahaz) [inline apps/condo/domains/news/utils/accessSchema/index.js:18]: why not to use type here?

**Review by pahaz — APPROVED** [review]

File: apps/condo/domains/news/schema/NewsItemScope.test.js (hunk 3 above)
Comment (AleX83Xpert) [reply apps/condo/domains/news/schema/NewsItemScope.test.js:321]: Error details are ok. The test name was wrong. Fixed.

File: apps/condo/domains/news/schema/NewsItemScope.js (hunk 2 above)
Comment (AleX83Xpert) [reply apps/condo/domains/news/schema/NewsItemScope.js:83]: Moved to the field's hook

**Pull request merged**